data-service/src/unit-tests/java/uk/gov/gchq/palisade/service/data/domain/DomainConverterTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.data.domain;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import javax.persistence.AttributeConverter;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DomainConverterTest {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    static class DomainConvertersSource implements ArgumentsProvider {
+        @Override
+        public Stream<? extends Arguments> provideArguments(final ExtensionContext extensionContext) throws Exception {
+            return Stream.of(
+                    Arguments.of(new ContextConverter(MAPPER), DomainTestData.CONTEXT),
+                    Arguments.of(new LeafResourceConverter(MAPPER), DomainTestData.LEAF_RESOURCE),
+                    Arguments.of(new RulesConverter(MAPPER), DomainTestData.RULES),
+                    Arguments.of(new UserConverter(MAPPER), DomainTestData.USER)
+            );
+        }
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(DomainConvertersSource.class)
+    <T> void testContextConverterIsConsistent(final AttributeConverter<T, String> converter, final T object) {
```

Comment:
update name of test to reflect it works on all domain objects not just Context object and the same for the rest of the tests in this file